.github/workflows/psammead-change-scanner.yml

```diff
@@ -17,7 +17,9 @@ jobs:
           persist-credentials: false
 
       - name: Install Node Modules
-        run: yarn ci
+        run: |
+          rm -rf node_modules
+          yarn install --immutable --check-cache
```

Comment:
Added cache checking for workflows due to this [suggestion](https://yarnpkg.com/features/zero-installs#does-it-have-security-implications). Would be better if we could create a workflow trigger that would check the cache based on the PR authors permissions in the repo, not sure if that's possible?

Reply:
Double check why yarn user scripts don't run when no node_modules folder is available

Reply:
From looking at where this error is thrown from within the Yarn source code [here](https://github.com/yarnpkg/berry/blob/ebb09ae70be905fa4fa5c3fb4a73e3b1af8bf0e9/packages/plugin-nm/sources/NodeModulesLinker.ts#L51). There doesn't seem to be much of an explanation for why it was added. I have found an issue raising this as a bug on the Yarn Berry repo https://github.com/yarnpkg/berry/issues/2701#issuecomment-814462813, with a comment explaining the reason it works like this, is because Yarn exposes all dependancy binaries, before running user scripts, even if a user script installs these binaries

Reply:
> From looking at where this error is thrown from within the Yarn source code [here](https://github.com/yarnpkg/berry/blob/ebb09ae70be905fa4fa5c3fb4a73e3b1af8bf0e9/packages/plugin-nm/sources/NodeModulesLinker.ts#L51). There doesn't seem to be much of an explanation for why it was added. I have found an issue raising this as a bug on the Yarn Berry repo https://github.com/yarnpkg/berry/issues/2701#issuecomment-814462813, with a comment explaining the reason it works like this, is because Yarn exposes all dependancy binaries before running user scripts, even if a user script installs those binaries

I'm not sure I entirely understand it but for now I'd consider introducing a little `yarn-install-ci.sh` script to encapsulate the commands and reference them in all the actions. Out of interest why do we need to delete the node modules directory (`rm -rf node_modules`), isn't every github action ran in a clean docker instance?